CMakeLists.txt

```diff
@@ -30,8 +30,8 @@ cmake_minimum_required(VERSION 3.13.1)
 project(openthread)
 
 option(OT_BUILD_EXECUTABLES "Build executables" ON)
-option(OT_BUILTIN_MBEDTLS "Enable builtin mbedTLS" ON)
 option(OT_COVERAGE "enable coverage" OFF)
+set(OT_EXTERNAL_MBEDTLS "" CACHE STRING "Specify external mbedtls library")
```

Comment:
Can this be OT_MBEDTLS with default value mbedtls. And use this variable instead of adding an interface target?

Reply:
We need a way to tell whether the user has set `OT_EXTERNAL_MBEDTLS` or not, so the default value has to be an invalid target name , such as `""`.

An [ALIAS target](https://cmake.org/cmake/help/latest/command/add_library.html#alias-libraries) should be more suitable for our usage case. But the `OT_EXTERNAL_MBEDTLS` target itself could be an `ALIAS target` and cmake doesn't allow ALIAS target on aliass.

Reply:
So how about use OT_MBEDTLS as an internal variable whose value is mbedtls and can be overridden by OT_EXTERNAL_MBEDTLS?